### PR TITLE
Make `req_pool_indices` on CPU

### PR DESCRIPTION
**Pull request by hnyls2002**
The allocation of `req_pool_indices` can be put on the CPU by just using python list. And we can reduce the GPU CPU sync overhead here.

### CODE REVIEW

**Comment by zhyncs** [comment]
The unit test failed. Does this improve performance?

**Comment by hnyls2002** [comment]
> The unit test failed. Does this improve performance?

I believe there is a minor improvement in the performance, but it can help us clean the code.

File: python/sglang/srt/managers/schedule_batch.py [hunk 1]
```diff
@@ -321,16 +321,54 @@ def init_new(cls, reqs, req_to_token_pool, token_to_kv_pool, tree_cache):
             return_logprob=return_logprob,
         )
 
+    def batch_size(self):
+        return len(self.reqs) if self.reqs is not None else 0
```
Comment (zhyncs) [inline python/sglang/srt/managers/schedule_batch.py:328]: Why encapsulate the previous `len(self.reqs)` in a function and include a None check? Why wasn't this problematic previously?

Comment (hnyls2002) [reply python/sglang/srt/managers/schedule_batch.py:328]: No, the current implementation and the previous one are both well with `len(reqs)`.

Comment (zhyncs) [reply python/sglang/srt/managers/schedule_batch.py:328]: ok

**Review by Ying1123 — APPROVED** [review]

**Pull request merged**